### DISCUSSION TITLE
Help me understand InvalidValidationError

### DISCUSSION THREAD

**Discussion by warvariuc** [go-playground/validator#888] [Q&A]
I am trying to validate a slice of some types. When I am checking for the error, I see `validator: (nil []map[string]interface {})` which I don't understand. What does this error mean in this example?

```
package main

import (
	"fmt"

	"github.com/go-playground/validator/v10"
)

func main() {
	validate := validator.New()
	data := make([]map[string]interface{}, 0)
	err := validate.Struct(data)
	if err != nil {
		if _, ok := err.(*validator.InvalidValidationError); ok {
			fmt.Printf("%+v", err)
		}
	}

}

```

**Comment by warvariuc** [comment]
I think I understand the issue. The validation works only on structs. I was imagining that it goes into slice items and map values (why not?)... The error message is not helpful.

**Reply by deankarn** [reply]
`validate.Var(..)` was added to handle other types after `validate.Struct` was created, which is the most common use case.

**Reply by fvitan** [reply]
> `validate.Var(..)` was added to handle other types after `validate.Struct` was created, which is the most common use case.

Yes,, but in that case the error messages are quite wrong:
e.g.: `"Key: '' Error:Field validation for '' failed on the 'hexadecimal' tag"`
